src/optimum/nvidia/models/llama.py

```diff
@@ -121,20 +100,24 @@ def convert(
             model.layers[idx].post_layernorm.weight.value = post_attn_ln
 
             # Self attention layer
-            qkv = qkv_packed_layers[idx]
-            q, k, v = qkv
+            qkv_weights, qkv_bias = qkv_packed_layers[idx]
+            q_weight, k_weight, v_weight = qkv_weights
+
+            if qkv_bias is not None:
+                raise ValueError("TensorRT-LLM's Llama does not support query/key/value projection bias.")
 
             # Shard
-            if not config.use_multi_head_attention:  # TODO: support GQA
+            # TODO: support GQA
+            if not config.use_multi_head_attention:
                 wq, wk, wv = (
-                    shard(q, rank, shard_info.tp_size, axis=0),
-                    shard(k, rank, shard_info.tp_size, axis=0),
-                    shard(v, rank, shard_info.tp_size, axis=0),
+                    shard(q_weight, rank, shard_info.tp_size, axis=0),
```

Comment:
we explicitly name the variable `q_weight` as `qkv_packed_layers` now yields both the weight and bias, and explicitly error out in case some q/k/v bias is not None, which may be the case in transformers: https://github.com/huggingface/transformers/blob/bb6aa8bc5ff8537f58c4b6ac80611101ba556226/src/transformers/models/llama/modeling_llama.py#L315-L318